src/main/java/de/numcodex/sq2cql/model/structured_query/TimeRestriction.java

```diff
@@ -30,16 +31,18 @@ public static TimeRestriction create(@JsonProperty("afterDate") String afterDate
         if (afterDate == null && beforeDate == null) {
             return null;
         } else if (afterDate == null) {
-            return TimeRestriction.of(LocalDate.of(1900, 1, 1), LocalDate.parse(beforeDate));
+            return TimeRestriction.of(LocalDate.MIN, LocalDate.parse(beforeDate));
         } else if (beforeDate == null) {
-            return TimeRestriction.of(LocalDate.parse(afterDate), LocalDate.of(2040, 1, 1));
+            return TimeRestriction.of(LocalDate.parse(afterDate), LocalDate.MAX);
         } else {
             return TimeRestriction.of(LocalDate.parse(afterDate), LocalDate.parse(beforeDate));
         }
     }
 
     public static TimeRestriction fromJsonNode(JsonNode node) {
-        return TimeRestriction.create(node.get("afterDate").asText(), node.get("beforeDate").asText());
+        final var afterDate = Optional.ofNullable(node.get("afterDate")).map(JsonNode::asText).orElse(null);
```

Comment:
I would not use `final` on local variables, because it should be obvious not to reassign them.